pygmt/helpers/tempfile.py

```diff
@@ -154,14 +155,8 @@ def tempfile_from_geojson(geojson):
             # Other 'geo' formats which implement __geo_interface__
             import json
 
-            import fiona
-
-            with fiona.Env():
-                jsontext = json.dumps(geojson.__geo_interface__)
-                # Do Input/Output via Fiona virtual memory
-                with fiona.io.MemoryFile(file_or_bytes=jsontext.encode()) as memfile:
-                    geoseries = gpd.GeoSeries.from_file(filename=memfile)
-                    geoseries.to_file(**ogrgmt_kwargs)
+            jsontext = json.dumps(geojson.__geo_interface__)
+            gpd.read_file(io.StringIO(jsontext)).to_file(**ogrgmt_kwargs)
```

Comment:
Why is codecov reporting that these two lines are not covered by tests? Also, should we temporarily set `engine='fiona'` here to prevent `ValueError: The 'schema' argument is not supported with the 'pyogrio' engine` in the GMT Dev Tests?

```suggestion
            jsontext = json.dumps(geojson.__geo_interface__)
            gpd.read_file(filename=io.StringIO(jsontext), engine="fiona").to_file(**ogrgmt_kwargs)
```

Reply:
> Also, should we temporarily set `engine='fiona'` here to prevent `ValueError: The 'schema' argument is not supported with the 'pyogrio' engine` in the GMT Dev Tests?

It won't work. `fiona` is not installed in the GMT Dev Tests workflow, because it's no longer a required dependency for geopandas 1.x.

Reply:
> Why is codecov reporting that these two lines are not covered by tests?

Not sure what happened, but now the coverage reports look good.

Reply:
> > Also, should we temporarily set `engine='fiona'` here to prevent `ValueError: The 'schema' argument is not supported with the 'pyogrio' engine` in the GMT Dev Tests?
> 
> It won't work. `fiona` is not installed in the GMT Dev Tests workflow, because it's no longer a required dependency for geopandas 1.x.

I meant to install `fiona` temporarily on GMT Dev Tests, so that the test pass for now with the change, but we'll need to look at how to handle this except-branch without `fiona` before PyGMT v0.13.0 is released. Also wondering if we should open an issue upstream to `geopandas` on what would be the recommended way forward if the 'schema' argument is not supported with `pyogrio`.

Reply:
It turns out `pyogrio` is "smarter" than fiona. 

In https://github.com/GenericMappingTools/pygmt/issues/2497#issuecomment-1614418132, I mentioned that 

> I thought world["pop_est_int"] = world.pop_est_round.astype("int32") should work, but it doesn't.

Changing the dtype directly doesn't work with fiona, but works well with pyorgio!.

In 2fb635bbf3fb3f92d88105875c62b02507a3f958, I've added the if-test to check if it's geopandas v0.x or v1.x. For geopandas v0.x, we use the old codes which pass the updated `schema` parameter. For geopandas v1.x, we change the column dtype when necessary.